### PR TITLE
ORLP-437 Search 

### PR DESCRIPTION
**Pull request by Anblak**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -186,6 +186,12 @@
             <artifactId>snakeyaml</artifactId>
             <version>1.18</version>
         </dependency>
+        <!--&lt;!&ndash; JDBC SPY&ndash;&gt;-->
```
Comment (VovaChornyi) [inline pom.xml]: maybe you can remove this lines

File: src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.softserve.academy.spaced.repetition.controller;
 
 
-import com.softserve.academy.spaced.repetition.controller.dto.impl.NewAccountPasswordDTO;
+import com.softserve.academy.spaced.repetition.controller.dto.simpleDTO.NewAccountPasswordDTO;
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java]: Why did you rename or move dtos?

File: src/main/java/com/softserve/academy/spaced/repetition/controller/SearchController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    @Autowired
+    private DeckService deckService;
+
+
+    @GetMapping(value = "search/{searchString}")
+    @ResponseStatus(HttpStatus.OK)
+    public List<String> getLinksFromSearch(@PathVariable String searchString) {
+        List<String> links = new ArrayList<>();
+        courseService.findCoursesId(searchString).forEach(deckId -> links.add(linkTo(methodOn(DeckController.class)
+                .getDeckById(deckId.longValueExact())).withSelfRel().getHref()));
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/SearchController.java]: What we get here, deck or course?
Maybe `.getCourseById`

File: src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java [hunk 4]
```diff
@@ -58,9 +58,9 @@
 
     public HttpHeaders getAuthHeaders(String email, String password, String captcha, Device device) throws UserStatusException {
         ReCaptchaResponseDto reCaptchaResponseDto = reCaptchaApiService.verify(captcha);
-        if (!reCaptchaResponseDto.isSuccess()) {
-            throw new BadCredentialsException("reCaptcha");
-        }
+//        if (!reCaptchaResponseDto.isSuccess()) {
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java]: remove this comments

File: wercker.yml [hunk 5]
```diff
@@ -12,6 +12,12 @@ build:
         code: |
           mkdir container/  
           cp ./target/*.jar container/
+  # Notify about the step status to slack
+  after-steps:
+    - slack-notifier:
+        url: $SLACK_URL
+        channel: notifications
+        username: Wercker Event Notifier
```
Comment (MoGUREV) [inline wercker.yml]: Did you change this file?

File: src/main/java/com/softserve/academy/spaced/repetition/service/cacheScheduler/CacheScheduler.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(CacheScheduler.class);
+
+    @Autowired
+    private CategoryService categoryService;
+
+    @Override
+    public void run(String... args) throws Exception {
+        logger.info("Started adding categories to cache...");
+        categoryService.getTopCategory();
```
Comment (VovaChornyi) [inline src/main/java/com/softserve/academy/spaced/repetition/service/cacheScheduler/CacheScheduler.java]: you add categories to cashe, but what about decks

File: src/main/java/com/softserve/academy/spaced/repetition/service/impl/CategoryServiceImpl.java [hunk 7]
```diff
@@ -20,41 +21,44 @@
...
 
     @Autowired
     private CategoryRepository categoryRepository;
     @Autowired
     private ImageService imageService;
 
+
     @Override
-    @Transactional
+    @Cacheable(ALL_CATEGORIES_CACHE_NAME)
```
Comment (VovaChornyi) [inline src/main/java/com/softserve/academy/spaced/repetition/service/impl/CategoryServiceImpl.java]: did you write cashing only for category?
what about decks?

File: src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java (hunk 2 above)
Comment (Anblak) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java]: because it`s not a impl of DTO

File: pom.xml (hunk 1 above)
Comment (Anblak) [reply pom.xml]: mb))0)

File: src/main/java/com/softserve/academy/spaced/repetition/controller/SearchController.java (hunk 3 above)
Comment (Anblak) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/SearchController.java]: oh
Yeap

File: src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java (hunk 4 above)
Comment (Anblak) [reply src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java]: no ^_^

File: wercker.yml (hunk 5 above)
Comment (Anblak) [reply wercker.yml]: it`s devops changes, pls see commits in pull request

File: src/main/java/com/softserve/academy/spaced/repetition/service/cacheScheduler/CacheScheduler.java (hunk 6 above)
Comment (Anblak) [reply src/main/java/com/softserve/academy/spaced/repetition/service/cacheScheduler/CacheScheduler.java]: decks cache will be much bigger

File: src/main/java/com/softserve/academy/spaced/repetition/service/impl/CategoryServiceImpl.java (hunk 7 above)
Comment (Anblak) [reply src/main/java/com/softserve/academy/spaced/repetition/service/impl/CategoryServiceImpl.java]: decks cache will be much bigger

File: src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java (hunk 2 above)
Comment (MoGUREV) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/RegistrationController.java]: Okay, i agree, but was it a part of your task?

File: src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java (hunk 4 above)
Comment (MoGUREV) [reply src/main/java/com/softserve/academy/spaced/repetition/security/service/AuthenticationRestService.java]: So, if it is necessary, why did you put comments?
